### PR TITLE
Tests for Personal ID Signup Parsers + More Validations on mandatory fields

### PR DESCRIPTION
**Pull request by shubham1g5**
## Product Description

https://dimagi.atlassian.net/browse/CCCT-1871

Unit tests on Network Parsers + Some more validations on mandatory fields inside Parsers. 

## Labels and Review

- [ ] Do we need to enhance the manual QA test coverage ? If yes, the "QA Note" label is set correctly
- [ ] Does the PR introduce any major changes worth communicating ? If yes, the "Release Note" label is set and a "Release Note" is specified in PR description.
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: app/src/org/commcare/connect/network/connectId/parser/ConfirmBackupCodeResponseParser.java [hunk 1]
```diff
@@ -18,15 +20,28 @@ public class ConfirmBackupCodeResponseParser implements PersonalIdApiResponsePar
...
+
         if (json.has("attempts_left")) {
             sessionData.setAttemptsLeft(json.getInt("attempts_left"));
         }
         if (json.has("error_code")) {
             sessionData.setSessionFailureCode(json.getString("error_code"));
         }
-        sessionData.setOauthPassword(JsonExtensions.optStringSafe(json, "password", null));
         sessionData.setInvitedUser(json.optBoolean("invited_user", false));
     }
```
Comment (conroy-ricketts) [inline app/src/org/commcare/connect/network/connectId/parser/ConfirmBackupCodeResponseParser.java]: Same comment here - this is great!

File: app/unit-tests/src/org/commcare/connect/network/connectId/parser/AddOrVerifyNameParserTest.kt [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        // Act
+        parser.parse(json, sessionData)
+
+        // Assert
+        assertEquals(true, sessionData.accountExists)
+        assertEquals(
+            "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+M9QDwADhgGAWjR9awAAAABJRU5ErkJggg==",
+            sessionData.photoBase64,
+        )
+    }
```
Comment (conroy-ricketts) [inline app/unit-tests/src/org/commcare/connect/network/connectId/parser/AddOrVerifyNameParserTest.kt]: _**Optional: **_ We could extract the long string into a variable

```suggestion
    @Test
    fun testParseCompleteValidResponse() {
        // Arrange
        val photoBase64 =
            "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+M9QDwADhgGAWjR9awAAAABJRU5ErkJggg=="
        val json =
            JSONObject().apply {
                put("account_exists", true)
                put("photo", photoBase64)
            }

        // Act
        parser.parse(json, sessionData)

        // Assert
        assertEquals(true, sessionData.accountExists)
        assertEquals(photoBase64, sessionData.photoBase64)
    }
```

File: app/unit-tests/src/org/commcare/connect/network/connectId/parser/ReportIntegrityResponseParserTest.kt
Comment (conroy-ricketts) [inline app/unit-tests/src/org/commcare/connect/network/connectId/parser/ReportIntegrityResponseParserTest.kt]: Is it possible to add some more test cases to this class? For example, is it practical to add tests for scenarios that would throw an `IOException` or a `JSONException`? Or another test for when the request ID is `null`?

File: app/unit-tests/src/org/commcare/connect/network/connectId/parser/StartConfigurationResponseParserTest.kt [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+                // Missing other fields
+            }
+
+        // Act
+        parser.parse(json, sessionData)
+
+        // Assert
+        assertEquals("biometric", sessionData.requiredLock)
+        assertEquals("partial-token", sessionData.token)
+        assertFalse(sessionData.demoUser ?: true) // optBoolean returns false when not set
```
Comment (conroy-ricketts) [inline app/unit-tests/src/org/commcare/connect/network/connectId/parser/StartConfigurationResponseParserTest.kt]: **_Optional:_** In the comment, it may be worth mentioning that `demoUser` is nullable here too like on line 71 and that's the reason for ` ?: true`

**Review by conroy-ricketts — COMMENTED** [review]
These unit tests are great! I think that the comments are fantastic

File: app/unit-tests/src/org/commcare/connect/network/connectId/parser/AddOrVerifyNameParserTest.kt [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+                )
+            }
+
+        // Act
+        parser.parse(json, sessionData)
+
+        // Assert
+        assertEquals(true, sessionData.accountExists)
+        assertEquals(
+            "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+M9QDwADhgGAWjR9awAAAABJRU5ErkJggg==",
```
Comment (OrangeAndGreen) [inline app/unit-tests/src/org/commcare/connect/network/connectId/parser/AddOrVerifyNameParserTest.kt]: Nit: Is there a reason not to store this as a const string so it isn't copy-pasted here and above?

File: app/src/org/commcare/connect/network/connectId/parser/ConfirmBackupCodeResponseParser.java [hunk 5]
```diff
@@ -18,15 +20,28 @@ public class ConfirmBackupCodeResponseParser implements PersonalIdApiResponsePar
...
     public void parse(JSONObject json, PersonalIdSessionData sessionData) throws JSONException {
-        sessionData.setPersonalId(JsonExtensions.optStringSafe(json, "username", null));
-        sessionData.setDbKey(JsonExtensions.optStringSafe(json, "db_key", null));
+        String username = JsonExtensions.optStringSafe(json, "username", null);
+        String dbKey = JsonExtensions.optStringSafe(json, "db_key", null);
+        String password = JsonExtensions.optStringSafe(json, "password", null);
+
+        Objects.requireNonNull(username);
+        Objects.requireNonNull(dbKey);
+        Objects.requireNonNull(password);
```
Comment (Jignesh-dimagi) [inline app/src/org/commcare/connect/network/connectId/parser/ConfirmBackupCodeResponseParser.java]: Do we want to add the text here as reason i.e. Objects.requireNonNull(username,"username cannot be null");

File: app/src/org/commcare/connect/network/connectId/parser/ConfirmBackupCodeResponseParser.java [hunk 6]
```diff
@@ -18,15 +20,28 @@ public class ConfirmBackupCodeResponseParser implements PersonalIdApiResponsePar
...
+        if (username.isEmpty() || dbKey.isEmpty() || password.isEmpty()) {
+            throw new IllegalStateException(
```
Comment (Jignesh-dimagi) [inline app/src/org/commcare/connect/network/connectId/parser/ConfirmBackupCodeResponseParser.java]: PersonalIdApiHandler doesn't handle `IllegalStateException` so application might crash here.

File: app/src/org/commcare/connect/network/connectId/parser/CompleteProfileResponseParser.java [hunk 7]
```diff
@@ -20,9 +20,21 @@ public class CompleteProfileResponseParser implements PersonalIdApiResponseParse
...
-        sessionData.setOauthPassword(Objects.requireNonNull(JsonExtensions.optStringSafe(json, "password", null)));
+        String username = JsonExtensions.optStringSafe(json, "username", null);
+        String dbKey = JsonExtensions.optStringSafe(json, "db_key", null);
+        String password = JsonExtensions.optStringSafe(json, "password", null);
+
+        Objects.requireNonNull(username);
+        Objects.requireNonNull(dbKey);
+        Objects.requireNonNull(password);
+        if (username.isEmpty() || dbKey.isEmpty() || password.isEmpty()) {
+            throw new IllegalStateException(
```
Comment (Jignesh-dimagi) [inline app/src/org/commcare/connect/network/connectId/parser/CompleteProfileResponseParser.java]: Same as https://github.com/dimagi/commcare-android/pull/3402/files#r2498594790

File: app/unit-tests/src/org/commcare/connect/network/connectId/parser/AddOrVerifyNameParserTest.kt [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+        // Act
+        parser.parse(json, sessionData)
+
+        // Assert
+        assertEquals(false, sessionData.accountExists) // optBoolean returns false for null
+        assertNull(sessionData.photoBase64)
+    }
+
+    @Test
+    fun testParseWithLargePhotoData() {
```
Comment (Jignesh-dimagi) [inline app/unit-tests/src/org/commcare/connect/network/connectId/parser/AddOrVerifyNameParserTest.kt]: Not sure if this test is different from other test except the string length
